src/app/features/rpc-transaction-request/hooks/use-rpc-broadcast-stacks-transaction.ts

```diff
@@ -0,0 +1,73 @@
+import { useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import type { StacksTransactionWire } from '@stacks/transactions';
+
+import type { Money } from '@leather.io/models';
+import { type RpcMethodNames, createRpcSuccessResponse } from '@leather.io/rpc';
+import { isString } from '@leather.io/utils';
+
+import { logger } from '@shared/logger';
+import { RouteUrls } from '@shared/route-urls';
+
+import { useRpcRequestParams } from '@app/common/hooks/use-rpc-request-params';
+import { useStacksBroadcastTransaction } from '@app/common/transactions/stacks/use-stacks-broadcast-transaction';
+import { useSignStacksTransaction } from '@app/store/transactions/transaction.hooks';
+
+import { useRpcTransactionRequestContext } from '../rpc-transaction-request.context';
+
+interface UserSetTxOptions {
+  fee: Money | null;
+  nonce: number | string;
+}
+
+export function useRpcBroadcastStacksTransaction<T>(method: RpcMethodNames) {
```

Comment:
can we be more specific here in what type can be passed, unclear to me as reader

e.g. `<T extends SomeTypeMatchingAllVariants>`